### PR TITLE
add s3_key to story background with proxy and color but no s3_key

### LINKED ISSUE #691: correctly process story background with s3_key and additional payload + add test

### PR DESCRIPTION
**Pull request by akhrail1996**
in parallel, I will fix the frontend so it submits s3_key instead of a proxy

closes #691

### CODE REVIEW

File: lib/t_web/channels/profile_channel.ex [hunk 1]
```diff
@@ -107,12 +107,15 @@ defmodule TWeb.ProfileChannel do
   end
 
   defp replace_photo_url_with_s3key(%{"s3_key" => _} = background), do: background
-  defp replace_photo_url_with_s3key(%{"color" => _} = bg), do: bg
```
Comment (akhrail1996) [inline lib/t_web/channels/profile_channel.ex:110]: the issue was this matching the story background with "proxy" + position (incl. "color") -> not getting s3_key

**Review by ruslandoga — CHANGES_REQUESTED** [review]
I'm missing what exactly this is solving. Can it be solved similar to stickers?

1. fix the client
2. remove `replace_photo_url_with_s3key`
3. story changeset function should validate/restore s3_key for background similar to labels
4. cleanup db

File: lib/t_web/channels/profile_channel.ex [hunk 2]
```diff
@@ -107,12 +107,15 @@ defmodule TWeb.ProfileChannel do
...
 
-  defp replace_photo_url_with_s3key(%{"proxy" => proxy_url}) do
-    %{"s3_key" => s3_key_from_proxy_url(proxy_url)}
+  defp replace_photo_url_with_s3key(%{"proxy" => proxy_url} = background) do
```
Comment (ruslandoga) [inline lib/t_web/channels/profile_channel.ex:111]: what's in `background`?

**Comment by ruslandoga** [comment]
> I'm missing what exactly this is solving.

Is the actual change on in the order of the function clauses?

File: lib/t_web/channels/profile_channel.ex (hunk 2 above)
Comment (akhrail1996) [reply lib/t_web/channels/profile_channel.ex:111]: e.g. https://github.com/getsince/test3/pull/693/files#diff-4b57c8e03ec7e4456f02252bb310d0189cf01a8285d6e238c9698a5a78c2bc57R357

**Comment by akhrail1996** [comment]
> Can it be done similar to stickers?

yes, will do. good idea, thanks!


> Is the change only in the order of the function clauses? s3_key > color > proxy becomes s3_key > proxy > color?

Also: https://github.com/getsince/test3/pull/693/files#diff-cd42229fa05d0e19bbee9514e6e6d57b07cb357bd02acf85336fb421332f8f41L113

**Comment by ruslandoga** [comment]
@akhrail1996 why zoom background?

**Comment by akhrail1996** [comment]
> @akhrail1996 why zoom background? I thought you were rasterising it on the client. Otherwise imgproxy is useless.

to allow resized (non-full screen) backgrounds:
image

full-screen (non-resized) background still do not have 'zoom'

**Comment by ruslandoga** [comment]
zoom > 1.0 -> crop, don't zoom
zoom < 1.0 -> update imgproxy scales

**Comment by akhrail1996** [comment]
> zoom > 1.0 -> crop, don't zoom

not nessecarily crop, for example:

no zoom (1.0)
image

zoom > 1.0
image

> zoom < 1.0 -> update imgproxy scales when picking [width bucket](https://github.com/getsince/test3/blob/225acee9c08a9d9f01a4ca5c67d9b5a69c43272d/lib/t/media.ex#L353)

👍, https://github.com/getsince/test5/issues/264

**Comment by ruslandoga** [comment]
@akhrail1996 are prev uploads assumed to be zoom = 1.0, does it mean the photos int he stories are not `fit` instead of `filled`?

and how does it work right now? If s3_key is not saved, what do the users see, proxy or color?

**Comment by akhrail1996** [comment]
> @akhrail1996 are prev uploads assumed to be zoom = 1.0

no, these are separate. all the prev uploads are ".scaleAspectFill"

> does it mean the photos int he stories are not `fit` instead of `filled`?

all the current uploads are '.scaleAspectFit'

> and how does it work right now? If s3_key is not saved, what do the users see, proxy or color?

proxy

**Comment by ruslandoga** [comment]
> all the current uploads are '.scaleAspectFit', regardless of zoom

how are they separated from the prev ones?

**Comment by akhrail1996** [comment]
> how are they separated from the prev ones?

prev ones do not have 'color' property (average color of the image)

**Pull request closed without merging**